nltk/tokenize/casual.py

```diff
@@ -156,7 +156,7 @@
 
     # Remaining word types:
     r"""
-    (?:[a-z][a-z'\-_]+[a-z])       # Words with apostrophes or dashes.
+    (?:\w[\w'\-_]+\w)              # Words with apostrophes or dashes.
```

Comment:
If the goal is to match alphabetic sequences optionally containing `-`, `'`, or `_` in the middle.

Another thought is that it might also be worth including the curly right single quote/apostrophe `’` at the beginning (_’twas_) or middle of the word. But I would guess that straight `'` at the beginning is more likely to be an open quote than an apostrophe.